camayoc/tests/remote/rho/test_scan.py

```diff
@@ -96,19 +96,19 @@ def test_scan(isolated_filesystem, profile):
             known_facts = {}
             for host in cfg['rho']['hosts']:
                 # find the facts for the scanned host we are inspecting
-                if host['ip'] == row['connection.host']:
+                if host['ip'] == row.get('connection.host'):
                     known_facts = host['facts']
                     break
             for fact in known_facts.keys():
                 try:
-                    assert (str(known_facts[fact]) in str(row[fact]))
+                    assert (str(known_facts[fact]) in str(row.get(fact)))
```

Comment:
One observation is that if the row does not have the fact then it will convert to the string `'None'` which won't match. But if the code get updated it may break

```python
 >>> str(None)
'None'

Reply:
Also why not use == here? It could have the false positive of 'abc' being in 'abcd'

Reply:
I was thinking that this would be more resilient to white space changes, when I was originally writing this and hand encoding the config file I think I had some issues with that.

Reply:
As per the 'None' issue,  am going to make a change and push it and we can see what you think.